enhancements/console/pinned-resources.md

```diff
@@ -0,0 +1,190 @@
+We can add `spec.customization.perspectives.pinnedResources` which lists default pre-pinned resources
+
+### Workflow Description
+
+To add default pre-pinned resources:
+
+```text
+├── spec
+│   ├── customization
+|       ├── perspectives
+|           ├── id
+|           ├── ...
+|           ├── pinnedResources
+|               └── group
+                └── version
+                └── resource
+└── ...
+```
+
+Each perspective conforms to the following schema :
+
+- `pinnedResources: operatorv1.PinnedResourceReference[]` , defines a list of pinned resources. The default pinned resources are contributed by the console UI extension plugin.  
+- `group: string` , defines the group of the resource.
+- `version: string` , defines the version of the resource.
+- `resource: string` , defines the type of the resource. It is the plural form of the resource kind in lowercase.
+
+If the `pinnedResources` list is empty, the default pinned resources contributed by the console UI extension plugin will be shown.
+If the `pinnedResources` list is non-empty, only the listed items will be shown if the user hasn't already customized the navigation items.
+If the `group`, `version` or `resource` is incorrect, the resource to be pinned will be ignored.
+
+Note: The Administrator perspective (id="admin") doesn't support pinned resources at the moment and can be excluded via CEL validation.
+
+
+### API Extensions
+
+Extend the existing `operator.openshift.io/v1` / `Console CRD` by adding `spec.customization.perspectives.pinnedResources`
+
+So the example YAML should look like this:
+
+To set default pre-pinned resources
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    perspectives:
+    - id: 'dev'
+      ...
+      pinnedResources:
+      - group: apps
+        version: v1
+        resource: deployments
+      - group: ""
+        version: v1
+        resource: configmaps
+    - ...
+  ...
+```
+
+### Risks and Mitigations
+
+**Risk 1:** A cluster admin configures an invalid combination of API/Group/Version
+Mitigation: Invalid combination of group, version and resource will not be shown to users. The cluster admin interface can show them with a warning.
+
+**Risk 2:** After a cluster is upgraded, a previously valid combination of API/Group/Version might get invalid
+Mitigation: Invalid combination of group, version and resource will not be shown to users. The cluster admin interface can show them with a warning.
+
+**Risk 3:** Currently, also if the user have not customized the pinned resources, the default settings are saved in user settings. After the update we can not determinate if the user touched them or not.
+Mitigation: For existing users where the saved pinned resources matches exactly the default pinned resources (from the perspective), the UI will show the customized default pinned resource (changed by the cluster admin).
```

Comment:
Won't this only catch the first time an admin changes the pinned resources? For instance, if the admin adds deployments one week and config maps the next... The config maps won't get added since the previous user setting was not the default (even if the user didn't customize anything).

Reply:
We have three buckets with pinned resources per perspective

1. Pinned resources from the perspective
2. Pinned resources from the customization
3. Pinned resources from the user settings

We should NOT save pinned resources until the user changes them manually (Add/Remove/Reorder). Unluckily we do this at the moment. So would recommend changing this, and then we need also to migrate the old data.

I hope we can do this while loading the data the first time like this:

1. If the user has pinned resources in the user settings:
    1. If these from the user settings are **_"deep" equal_** to the pinned resources from the perspective => delete the PR from user settings.
    2. Otherwise use the pinned resources from the user settings.
2. If there are pinned resources in the customization use them
3. Fallback to the pinned resources from the plugin definition.

We could also think about a more explicit migration, but I would keep it simple here and just compare both lists.

Reply:
@jerolimov's proposal here looks reasonable to me. We should update the enhancement to cover this.